src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TextDifferencing/TextDiffer.DiffEdit.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.PooledObjects;
+
+namespace Microsoft.AspNetCore.Razor.TextDifferencing;
+
+internal abstract partial class TextDiffer
+{
+    protected readonly struct DiffEdit
```

Comment:
I think I'm not following the purpose of this class because if it's supposed to represent an edit doesn't it need to have a "NewText" property for when it's inserting?

Reply:
This is essentially the same struct that was there before my work. It's used by the algorithm to keep track of differences between the source texts. At the end, there's another method ("ConsolidateEdits") that uses this to produce a list of `TextChange`.